Source/WebKit/Shared/ContextMenuContextData.cpp

```diff
@@ -90,6 +94,28 @@ void ContextMenuContextData::setImage(WebCore::Image* image)
 }
 #endif
 
+#if ENABLE(CONTEXT_MENU_QR_CODE_DETECTION)
+
+void ContextMenuContextData::setPotentialQRCodeNodeSnapshotImage(WebCore::Image* image)
+{
+    m_potentialQRCodeNodeSnapshotImage = ShareableBitmap::create(IntSize(image->size()), { });
+    auto graphicsContext = m_potentialQRCodeNodeSnapshotImage->createGraphicsContext();
+    if (!graphicsContext)
+        return;
+    graphicsContext->drawImage(*image, IntPoint());
+}
+
+void ContextMenuContextData::setPotentialQRCodeViewportSnapshotImage(WebCore::Image* image)
+{
+    m_potentialQRCodeViewportSnapshotImage = ShareableBitmap::create(IntSize(image->size()), { });
```

Comment:
what if `image` is `nullptr`?

Reply:
This is a private method that has a null check at the call site. I can add an `ASSERT` though.

Reply:
Actually, these can just be references.